### PR TITLE
Issue-441:Convert Pytest-CVP Cooling Status test case to Vane #441

### PR DESCRIPTION
**Pull request by paragpande24**
# Please include a summary of the changes

Converted test_system_hardware_cooling_status.py  as per the vane style guide along with that updated following files.
Use Test ID: NRFU6.2
test_definition.yaml: Updated test def with testcase  NRFU6.2
test_definition.yaml.j2: Updated J2 with testcase  NRFU6.2
pytest.ini: Added marker nrfu_test and system
nrfu_tests/nrfu_tests/test_system_hardware_cooling_status.py: Added testcase for NRFU6.2
masterdef.yaml: Updated master def file with global filter.
Test case should be skipped if vEOS : Added condition in test case.

# Any specific logic/part of code you need extra attention on

This testcase will verify the System cooling status should be 'coolingOk' in 'show system environment cooling' command output. If cooling status is not found test case will fail.

# Include the Issue number and link
#441 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU Testcase)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?
Unit Tests:

- Tested for all the verifying that the system status is ok..: Passed

[PASS-441.docx](https://github.com/aristanetworks/vane/files/12610068/PASS-441.docx)

- Tested if system status is not ok.: Failed. (with hardcoded output)
 
[441-FAIL-REPORT.docx](https://github.com/aristanetworks/vane/files/12606008/441-FAIL-REPORT.docx)

- Verified system status is ok: passed (With generate test definition True and J2 True )

[PASS-441-WITH-J2.docx](https://github.com/aristanetworks/vane/files/12610080/PASS-441-WITH-J2.docx)

   
- Test case skipped if device platform is vEOS.

[441-SKIP-REPORT.docx](https://github.com/aristanetworks/vane/files/12605997/441-SKIP-REPORT.docx)

- HTML reports:

[441-REPORT-HTML.zip](https://github.com/aristanetworks/vane/files/12610114/441-REPORT-HTML.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            dut(dict): details related to a particular DUT
+            tests_definitions(dict): test suite and test case parameters.
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        tops.expected_output = {}
+        tops.actual_output = {}
+        self.output = ""
+
+        # forming output message if test result is passed
+        tops.output_msg = "System cooling status is ok."
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:42]:
```suggestion
        tops.output_msg = "System cooling status is ok."

```

File: nrfu_tests/test_definition.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_system_hardware_cooling_status
+    test_id: NRFU6.2
+    description: Testcase for verification system cooling status.
+    show_cmd: show system environment cooling
+    expected_output: null # Forming expected output in test file.
+    test_criteria: System cooling status should be ok. # Setting test case’s pass / fail criteria for reporting
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:8]:
```suggestion
    test_criteria: System cooling status should be ok with expected ambient temperature. # Setting test case’s pass / fail criteria for reporting
```

File: nrfu_tests/test_system_hardware_cooling_status.py (hunk 1 above)
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:42]: as per comment in test def

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        try:
+            """
+            TS: Running 'show version' command on device and Skipping the test case
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:46]:
```suggestion
            TS: Running `show version` command on device and skipping the test case

```

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            for device if platform is vEOS.
+            """
+            version_output = dut["output"]["show version"]["json"]
+            logger.info(
+                "On device %s: Output of 'show version' command is: \n%s\n",
+                tops.dut_name,
+                version_output,
+            )
+            self.output += f"\n\nOutput of 'show version' command is: \n{version_output}"
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:55]:
```suggestion
            self.output += f"Output of 'show version' command is: \n{version_output}"

```

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+
+            self.output += f"\n\nOutput of {tops.show_cmd} command is: \n{output}"
+            system_cooling_status = output.get("systemStatus")
+            ambient_temperature = output.get("ambientTemperature")
+            tops.actual_output.update({"system_cooling_status": system_cooling_status})
+            tops.expected_output.update({"system_cooling_status": "coolingOk"})
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:77]: can we form this in starting of the test case ?

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            logger.info(
+                "On device %s: Output of 'show version' command is: \n%s\n",
+                tops.dut_name,
+                version_output,
+            )
+            self.output += f"\n\nOutput of 'show version' command is: \n{version_output}"
+
+            # Skipping testcase if device is vEOS.
+            if "vEOS" in version_output.get("modelName"):
+                pytest.skip(f"{tops.dut_name} is vEOS device, hence test skipped.")
```
Comment (hsisalgslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:59]:
```suggestion
                pytest.skip(f"{tops.dut_name} is vEOS device, hence test is skipped.")

```

**Review by hsisalgslab — APPROVED** [review]

File: nrfu_tests/test_definition.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_system_hardware_cooling_status
+    test_id: NRFU6.2
+    description: Testcase for verification system cooling status.
+    show_cmd: show system environment cooling
+    expected_output: null # Forming expected output in test file.
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_definition.yaml:7]: please update the following block with 
image

similar to this 
image

File: nrfu_tests/test_definition.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    test_criteria: System cooling status should be 'coolingOk' with expected ambient temperature. # Setting test case’s pass / fail criteria for reporting
+    report_style: modern # Setting report_style as modern If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
+    criteria: names
+    filter:
+      - BLFE1
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_definition.yaml:12]: image

Please update this block with what exactly this test case is all about and the verification parameters.

File: nrfu_tests/test_definition.yaml.j2 [hunk 9]
```diff
@@ -0,0 +1,11 @@
+- name: nrfu_tests
+  testcases:
+    - name: test_system_hardware_cooling_status
+      test_id: NRFU6.2
+      description: Testcase for verification system cooling status.
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_definition.yaml.j2:5]:
```suggestion
      description: Testcase for the verification of system cooling status.

```

File: nrfu_tests/test_definition.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_system_hardware_cooling_status
+    test_id: NRFU6.2
+    description: Testcase for verification system cooling status.
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_definition.yaml:5]:
```suggestion
    description:  Testcase for the verification of system cooling status.
```

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2023 Arista Networks, Inc.  All rights reserved.
+# Arista Networks, Inc. Confidential and Proprietary.
+
+"""
+Testcase for verification system cooling status.
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:5]: Testcase for the verification of system cooling status.

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    """
+
+    dut_parameters = tests_tools.parametrize_duts(TEST_SUITE, test_defs, dut_objs)
+    test_duts = dut_parameters["test_system_hardware_cooling_status"]["duts"]
+    test_ids = dut_parameters["test_system_hardware_cooling_status"]["ids"]
+
+    @pytest.mark.parametrize("dut", test_duts, ids=test_ids)
+    def test_system_hardware_cooling_status(self, dut, tests_definitions):
+        """
+        TD: Testcase for verification system cooling status.
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:31]: same as above

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+            self.output += f"Output of {tops.show_cmd} command is: \n{output}"
+            system_cooling_status = output.get("systemStatus")
+            ambient_temperature = output.get("ambientTemperature")
+            tops.actual_output.update({"system_cooling_status": system_cooling_status})
+
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = (
+                    "System cooling status is not ok.\nAmbient Temperature"
+                    f" is: {ambient_temperature} C"
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:81]: please use the following format for ambient temperature

image

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+            )
+
+            self.output += f"Output of {tops.show_cmd} command is: \n{output}"
+            system_cooling_status = output.get("systemStatus")
+            ambient_temperature = output.get("ambientTemperature")
+            tops.actual_output.update({"system_cooling_status": system_cooling_status})
+
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = (
+                    "System cooling status is not ok.\nAmbient Temperature"
```
Comment (geetanjalimanegslab) [inline nrfu_tests/test_system_hardware_cooling_status.py:80]:
```suggestion
                    "System cooling status is not ok. Ambient Temperature"

```

**Review by geetanjalimanegslab — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_definition.yaml [hunk 15]
```diff
@@ -0,0 +1,12 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_system_hardware_cooling_status
+    test_id: NRFU6.2
+    description: Testcase for the verification of system cooling status.
+    show_cmd: show system environment cooling
+    expected_output: null # Forming expected output in test file.
+    test_criteria: System cooling status should be 'coolingOk' with expected ambient temperature. # Setting test case’s pass / fail criteria for reporting
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_definition.yaml:8]: we are not going to verify the temp. Just the status should not coolingOk. If not ok then in observation we should print the temp

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+from vane.logger import logger
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_test
+@pytest.mark.system
+class SystemHardwareCoolingStatusTests:
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_system_hardware_cooling_status.py:19]:
```suggestion
class SystemCoolingStatusTests:

```

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+    @pytest.mark.parametrize("dut", test_duts, ids=test_ids)
+    def test_system_hardware_cooling_status(self, dut, tests_definitions):
+        """
+        TD: Testcase for the verification of system cooling status.
+        Args:
+            dut(dict): details related to a particular DUT
+            tests_definitions(dict): test suite and test case parameters.
+        """
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        tops.expected_output = {"system_cooling_status": "coolingOk"}
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_system_hardware_cooling_status.py:37]: put it in test def. There is no dynamic param then why its in python file

File: nrfu_tests/test_system_hardware_cooling_status.py [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+
+            self.output += f"Output of {tops.show_cmd} command is: \n{output}"
+            system_cooling_status = output.get("systemStatus")
+            ambient_temperature = output.get("ambientTemperature")
+            tops.actual_output.update({"system_cooling_status": system_cooling_status})
+
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = (
+                    "System cooling status is not ok. Ambient Temperature: "
+                    f"{format(ambient_temperature,'.2f')}C')"
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_system_hardware_cooling_status.py:81]: image
 Please remove this extra closing parenthesis

**Review by MaheshGSLAB — CHANGES_REQUESTED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

**Review by geetanjalimanegslab — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**